augur/util_support/color_parser.py

```diff
@@ -0,0 +1,31 @@
+from io import TextIOWrapper
+import functools
+from pkg_resources import resource_stream
+
+from augur.util_support.color_parser_line import ColorParserLine
+
+
+class ColorParser:
+    def __init__(self, *, mapping_filename, use_defaults=True):
+        self.mapping_filename = mapping_filename
+        self.use_defaults = use_defaults
+
+    @property
+    @functools.lru_cache()
+    def mapping(self):
+        colors = {}
+
+        if self.use_defaults:
+            with resource_stream("augur", "data/colors.tsv") as stream:
+                with TextIOWrapper(stream, "utf-8") as defaults:
+                    colors = {**colors, **self.parse_file(defaults)}
+
+        if self.mapping_filename:
+            with open(self.mapping_filename, encoding="utf-8") as mapping:
```

Comment:
Thank you for changing this logic to open the given filename without the initial check for whether the file is real or not! I would absolutely expect a missing file here to throw an error instead of a warning.

Would you mind also wrapping the `read_colors` calls in `augur export v1` and `augur export v2` with a try/except block that catches the `FileNotFound` exception raised when an invalid file is passed in and print the corresponding error to stderr and exit with a nonzero code as you recently did for the same type of error with the `read_metadata` function?

We really need a better general solution to this kind of error handling for augur, but for now this change will make life nicer for users on the command line.

Reply:
Updated.